amd_openvx_extensions/amd_nn/src/topk_layer.cpp

```diff
@@ -0,0 +1,294 @@
+
+    return VX_SUCCESS;
+}
+
+static vx_status VX_CALLBACK processTopKLayer(vx_node node, const vx_reference * parameters, vx_uint32 num)
+{
+    //get tensor dimensions
+    vx_size input_dims_0[4], input_dims_1[4], output_dims_0[4], output_dims_1[4];
+    vx_size num_of_dims;
+
+    vx_enum usage = VX_READ_ONLY;
+    vx_status status;
+    vx_map_id map_id;
+    vx_size stride[4];
+
+    //query and copy inputs
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DIMS, input_dims_0, sizeof(input_dims_0)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_NUMBER_OF_DIMS, &num_of_dims, sizeof(num_of_dims)));
+
+    float * ptr_input_0;
+    vx_size count_input_dims_0 = input_dims_0[0]*input_dims_0[1]*input_dims_0[2]*input_dims_0[3];
+    float *x_tensor = new float[count_input_dims_0]; 
+
+    status = vxMapTensorPatch((vx_tensor)parameters[0], num_of_dims, nullptr, nullptr, &map_id, stride, (void **)&ptr_input_0, usage, VX_MEMORY_TYPE_HOST, 0);
+    if(status)
+    {
+        std::cerr << "ERROR: vxMapTensorPatch() failed for input#1 (" << status << ")" << std::endl;
+        return -1;
+    }
+
+    memcpy(x_tensor, ptr_input_0, (count_input_dims_0*sizeof(float)));
+
+    status = vxUnmapTensorPatch((vx_tensor)parameters[0], map_id);
+
+    if(status) 
+    {
+        std::cerr << "ERROR: vxUnmapTensorPatch() failed for input#5 (" << status << ")" << std::endl;
+        return -1;
+    }
+
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_DIMS, input_dims_1, sizeof(input_dims_1)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_NUMBER_OF_DIMS, &num_of_dims, sizeof(num_of_dims)));
+
+    
+    int64_t * ptr_input_1;
+    vx_size count_input_dims_1 = input_dims_1[0];
+    int64_t *k_tensor = new int64_t[count_input_dims_1];
+
+    status = vxMapTensorPatch((vx_tensor)parameters[1], num_of_dims, nullptr, nullptr, &map_id, stride, (void **)&ptr_input_1, usage, VX_MEMORY_TYPE_HOST, 0);
+    if(status)
+    {
+        std::cerr << "ERROR: vxMapTensorPatch() failed for input#2 (" << status << ")" << std::endl;
+        return -1;
+    }
+
+    memcpy(k_tensor, ptr_input_1, (count_input_dims_1*sizeof(int64_t)));
+
+    status = vxUnmapTensorPatch((vx_tensor)parameters[1], map_id);
+    if(status) 
+    {
+        std::cerr << "ERROR: vxUnmapTensorPatch() failed for input#5 (" << status << ")" << std::endl;
+        return -1;
+    }
+
+    vx_int32 axis;
+    ERROR_CHECK_STATUS(vxCopyScalar((vx_scalar)parameters[2], &axis, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+
+    //opposite of onnx. Maybe have to remove for model. Works with GDF testing
+    axis = 3 - axis;
+    
+    //accessing the tensor from last dimension
+    if(axis > 3)
+    {
+        int count = 0, dims_idx = 0;
+        //get real dimensions. Ignoring the appended 1s
+        while(input_dims_0[dims_idx] == 1 && dims_idx <= 3)

```

Comment:
Codacy found an issue: [Array index 'dims_idx' is used before limits check.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5835780)